### PR TITLE
Add instructions to generate type hints

### PR DESCRIPTION
**Pull request by davisv7**
## Change Description
Use mypy to generate .pyi files as well. These files are useful for type hinting in IDEs.

## Steps to Test
Run commands as listed in guide.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: docs/grpc/python.md [hunk 1]
```diff
@@ -35,12 +35,11 @@ file in Python before you can use it to communicate with lnd.
     ```
 6. Compile the proto file
     ```shell
-    lnd $  python -m grpc_tools.protoc --proto_path=googleapis:. --python_out=. --grpc_python_out=. lightning.proto
+    lnd $  python -m grpc_tools.protoc --proto_path=googleapis:. --mypy_out=. --python_out=. --grpc_python_out=. router.proto
```
Comment (guggero) [inline docs/grpc/python.md:38]: Why change this to `router.proto`?

File: docs/grpc/python.md [hunk 2]
```diff
@@ -50,11 +49,11 @@ generate the python modules for them.
 For example, if you want to generate the RPC modules for the `Router` subserver
 (located/defined in `routerrpc/router.proto`), you need to run the following two
 extra steps (after completing all 6 step described above) to get the
-`router_pb2.py` and `router_pb2_grpc.py`:
+`router_pb2.py`, `router_pb2_grpc.py` and `router_pb2.pyi`:
 
 ```shell
 lnd $  curl -o router.proto -s https://raw.githubusercontent.com/lightningnetwork/lnd/master/lnrpc/routerrpc/router.proto
-lnd $  python -m grpc_tools.protoc --proto_path=googleapis:. --python_out=. --grpc_python_out=. router.proto
+lnd $  python -m grpc_tools.protoc --proto_path=googleapis:. --mypy_out=. --python_out=. --grpc_python_out=. lightning.proto
```
Comment (guggero) [inline docs/grpc/python.md:56]: Looks like these two lines were swapped.

**Review by guggero — COMMENTED** [review]
Thanks for the improvement! Haven't tested it but sounds good to me. Just need to revert the changes in the order of generating which file.

File: docs/grpc/python.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ file in Python before you can use it to communicate with lnd.
 3. Install dependencies (googleapis-common-protos is required due to the use of
   google/api/annotations.proto)
     ```shell
-    lnd $  pip install grpcio grpcio-tools googleapis-common-protos
+    lnd $  pip install grpcio grpcio-tools googleapis-common-protos mypy mypy-protobuff
```
Comment (bitromortac) [inline docs/grpc/python.md:23]: typo: `mypy-protobuff`->`mypy-protobuf`

**Review by bitromortac — COMMENTED** [review]
Cool idea, got picked up by my language server! A more minimal example seems to be

```
#!/bin/bash
python3 -m venv venv
source venv/bin/activate

git clone https://github.com/googleapis/googleapis.git

pip install --upgrade pip
pip install grpcio-tools mypy-protobuf

curl -o lightning.proto -s https://raw.githubusercontent.com/lightningnetwork/lnd/master/lnrpc/lightning.proto
python -m grpc_tools.protoc --proto_path=googleapis:. --mypy_out=. --mypy_grpc_out=. --python_out=. --grpc_python_out=. lightning.proto
```
This gets rid of the virtualenv dependency, installs all necessary dependencies automatically (`mypy` is not a dependency) and also generates `lightning_pb2_grpc.pyi`. Perhaps we could update it along those lines if this works for others as well?

**Comment by guggero** [comment]
Thanks, looks good. Waiting for the merge window to open again after the final version of `v0.17.0-beta` is now out.
In the meantime, can you please squash your commits into one? Or should we do that for you when merging the PR?

**Review by guggero — APPROVED** [review]
Looks good, going to squash and merge once @bitromortac has given thumbs up.

**Pull request merged**